hydra-node/golden/HydraNodeLog SimpleTx.json

```diff
@@ -1,350 +0,0 @@
-                        {
-                            "vkey": "53df8f812267bb9b04699777a6ff1b6fd313a3e4a5829e819f6a2b523310704c"
-                        },
-                        {
-                            "vkey": "14ddaf67c53862d4ad372d21306992ff8cbd46a5ce718a30b22e3dc1300c0f4c"
-                        },
-                        {
-                            "vkey": "daf0f57351dbf8af9cf9be2a0cc2be17d3517cfb7f9165ecaa24cb86599f8b5d"
-                        },
-                        {
-                            "vkey": "dcfc4a159ffe7f532208b49a998cb4d4416027e75640982433b88552fd906ccf"
-                        },
-                        {
-                            "vkey": "8704a3d2327f1784fd344268389707cd1a52f71ff471994991d55a471269d466"
-                        },
-                        {
-                            "vkey": "6753d20e068ad7d4da449e145f17991bc2135efb4b764bb8a140682c897afee1"
-                        },
-                        {
-                            "vkey": "7f2a8812a7ab37ceada7b5c2cb833491384c1b65d29239a20c3b327af79fea64"
-                        },
-                        {
-                            "vkey": "5b4d854a0ee7fa7075c96be0de21784871dd212a0a956bb2dda0697bdd6ab3e2"
-                        },
-                        {
-                            "vkey": "47177b3eca9e95e9f04b9a4cccf8362c41b4cf747ffed6ba1fced372b42e319a"
-                        },
-                        {
-                            "vkey": "83eedcc00b57c3634fc640b3d3c8f0a0696d89605f5007203cfd15ba32e7eda3"
-                        },
-                        {
-                            "vkey": "a27fe3f494dd5fb7edd56d90a5d4039fa08728f4c8aec875c222e693eb9b9da7"
-                        },
-                        {
-                            "vkey": "53c1c05bf738194647b90f503071d6c07ba26f3d6d1f8bbcc6cdadd3286e401f"
-                        }
-                    ],
-                    "tag": "PartiesMismatch"
-                }
-            ],
-            "tag": "Misconfiguration"
-        },
-        {
-            "by": {
-                "vkey": "54e1373ff14d0013bc9130dbb2347dbaf5f9d93fe35109a69e39f08a4d7f7d1a"
-            },
-            "outcome": {
-                "left": {
-                    "error": {
-                        "otherHeadId": "998bc89929dbad113ea20e4d8598de99",
-                        "ourHeadId": "e999d1a86b0d5baa5d615946e8dc7b77",
-                        "tag": "NotOurHead"
-                    },
-                    "tag": "Error"
-                },
-                "right": {
-                    "event": {
-                        "chainState": {
-                            "slot": 28
-                        },
-                        "contestationDeadline": "1864-06-07T14:40:39.769739526323Z",
-                        "tag": "HeadClosed"
-                    },
-                    "tag": "StateChanged"
-                },
-                "tag": "Combined"
```

Comment:
THIS exactly is the purpose of this PR. Make invalid states unrepresentable!

Reply:
You could have achieved the same result by just changing the `(<>)` semantics and have `Error` be an absorbing element, isn't it?

Reply:
It's actually done that way. I had briefly an `error` on the `<>`, but now it's just discarding. Maybe making it actually impossible to express on the producing site would be even better 🤔

The rest was a refactor on the outcome data constructors to have only valid results (on the usage site of Outcome).

Reply:
So you can't `Wait` and have some `Effect`s, eg. like a notification to clients?

Reply:
Currently you can't and seemingly we don't require it. I made the types only as powerful as we needed them.